plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/internal/RootResolvedTypes.java

```diff
@@ -131,7 +131,7 @@ protected void addTypeDiagnostic(XExpression expression, /* @Nullable */ TypeDat
 				if (expectedType != null) {
 					if (!expectedType.isPrimitiveVoid()) {
 						int conformanceFlags = getConformanceFlags(typeData, false);
-						if (!((conformanceFlags & ConformanceFlags.SUCCESS) != 0)) {
+						if ((conformanceFlags & ConformanceFlags.SUCCESS) == 0 || (conformanceFlags & ConformanceFlags.UNKNOWN_TYPE_PARTICIPATED) != 0) {
```

Comment:
Why would you create a conformance problem for unknown types? They should be flagged as missing on the type ref itself.